var/spack/repos/builtin/packages/hpl/package.py

```diff
@@ -30,6 +30,14 @@ class Hpl(AutotoolsPackage):
     depends_on("mpi@1.1:")
     depends_on("blas")
 
+    with when("@=2.3"):
+        depends_on("autoconf-archive", type="build")  # AX_PROG_CC_MPI
+        depends_on("autoconf", type="build")
+        depends_on("automake", type="build")
+        depends_on("m4", type="build")
+        depends_on("libtool", type="build")
+        force_autoreconf = True
```

Comment:
This is wrong. Only directives are affected by the `when` context manager, not class level attributes.

Reply:
Oops, fixed